### ISSUE TITLE
Some user are unable to login

### ISSUE THREAD

**Issue by mymasse** [nuxsmin/sysPass#967]
Fresh install of SysPass 2.1.15.17101701 with PhP version 7.0.30. Ldap configure with a bind user and a search base only (no group).

Check connection returns 3000+ users as expected. Some users can login some get an 'Internal Error' on the login screen. I've confirmed that these users show up in the Check Connection Results. Nothing in the httpd logs or the syspass.log in the config folder.

What can I do to troubleshoot this, this is preventing us from using SysPass

**Comment by nuxsmin** [comment]
Hello , did you check out the sysPass event log?. This issue could be caused by returning multiple results for a given cn, either for a group or user object.

**Comment by mymasse** [comment]
Is that the 'Duplicated user login/email' events? I do see some, how do i fix this?

**Comment by mymasse** [comment]
I would love to use a group to restrict access but everytime i try to put one the check configuration fails. In the event log I get 'Error while searching objects in base DN`.
All these LDAP settings are the same used on other application with success.

**Comment by nuxsmin** [comment]
Right, duplicated user's login or email will throw an error, since sysPass will avoid to store duplicated values for these fields.

The group filtering issue could be caused by a duplicated group too. Did you set a full DN for the group field?. If not, sysPass will lookup every group which matches, despite of the OU it is stored in, so it will fail when multiple groups are returned.

**Comment by mymasse** [comment]
So what's the solution? Fix the data in LDAP, how do I find what is duplicated

**Comment by nuxsmin** [comment]
You could search for those users which fail to log in and the group which you're using for filtering.

**Comment by mymasse** [comment]
OK I've downloaded JXplorer and connected to our LDAP with the same Bind user. If I search with the same 'search base' for uid=<user who can't login> I get only one result, what else can I do. I also don't specify a group.

**Comment by nuxsmin** [comment]
Please, perform a search with the given group.

You can get the search filter used by checking in sysPass event log.

**Comment by mymasse** [comment]
Like I mentioned I have not specified a group at all in sysPass. 

When users are unable to login the event doesn't list anything other than 'Duplicate user login/email' no filter is provided in the message.

**Comment by nuxsmin** [comment]
Sorry I missed your reply...

It seems that this error is thrown when it tries to add a new user to sysPass database, so it checks for a duplicated login or email in it.

**Comment by mymasse** [comment]
OK I found the problem, all our users have two mail attributes in LDAP and in syspass the email is entered as 'Array' so when a second user tries to login we get the duplicate error. Is it easy to pick the first email in the array?

**Comment by nuxsmin** [comment]
Hmm, nice, it should be done in line 525 of https://github.com/nuxsmin/sysPass/blob/master/inc/SP/Auth/Ldap/LdapBase.class.php but I think something is wrong because `count > 1` means that the entry has more than one entry so it should pick the first one in that piece of code but isn't, it's picking the first for `else` cases instead.

Please let me check out more deeply.
